File: frontend/src/api/skin.ts
import client from "./client";

/* ── Types matching backend SkinMeasurement schema ── */

export interface SkinMeasurementResponse {
  measurement_id: string;
  user_id: string;
  measured_at: string;
  hydration_score: number;
  elasticity_score: number;
  pore_score: number;
  wrinkle_score: number;
  pigmentation_score: number;
  overall_skin_score: number;
  classification_data: Record<string, unknown> | null;
  regression_data: Record<string, unknown> | null;
  image_url: string | null;
  capture_metadata: Record<string, unknown> | null;
  created_at: string;
}

export interface SkinMeasurementPayload {
  user_id: string;
  hydration_score: number;
  elasticity_score: number;
  pore_score: number;
  wrinkle_score: number;
  pigmentation_score: number;
  overall_skin_score: number;
  classification_data?: Record<string, unknown> | null;
  regression_data?: Record<string, unknown> | null;
  image_url?: string | null;
  capture_metadata?: Record<string, unknown> | null;
}

/** Save a skin measurement (auto-creates daily_skin_score on the backend). */
export async function saveMeasurement(
  payload: SkinMeasurementPayload,
): Promise<SkinMeasurementResponse> {
  const { data } = await client.post<SkinMeasurementResponse>(
    "/v1/skin/measure",
    payload,
  );
  return data;
}

/** Get the latest measurement for the current user. */
export async function getLatestMeasurement(): Promise<SkinMeasurementResponse | null> {
  const { data } = await client.get<SkinMeasurementResponse | null>(
    "/v1/skin/latest",
  );
  return data;
}

File: backend/app/schemas/skin.py
from datetime import datetime

from pydantic import BaseModel, Field


class SkinMeasurementCreate(BaseModel):
    user_id: str
    hydration_score: float = Field(..., ge=0, le=100)
    elasticity_score: float = Field(..., ge=0, le=1)
    pore_score: float = Field(..., ge=0, le=2600)
    wrinkle_score: float = Field(..., ge=0, le=50)
    pigmentation_score: float = Field(..., ge=0, le=350)
    overall_skin_score: int = Field(..., ge=0, le=100)
    classification_data: dict | None = None
    regression_data: dict | None = None
    image_url: str | None = Field(None, max_length=500)
    capture_metadata: dict | None = None


class SkinMeasurementRead(BaseModel):
    measurement_id: str
    user_id: str
    measured_at: datetime
    hydration_score: float
    elasticity_score: float
    pore_score: float
    wrinkle_score: float
    pigmentation_score: float
    overall_skin_score: int
    classification_data: dict | None = None
    regression_data: dict | None = None
    image_url: str | None = None
    capture_metadata: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}
